### ISSUE TITLE
Parsing Error from Named Function Expression in strict mode

### ISSUE THREAD

**Issue by danfike** [eslint/eslint#8378]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 6.9.2
* **npm Version:** 2.15.11

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`.eslintrc.json`:
```
{
    "env": {
        "node": true,
        "es6": true
    },
    "rules": {
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

I ran `eslint foo.js` on the following file `foo.js`...
```js
'use strict';

const _foo = function _foo () {
    return 15;
};

console.log('woot', _foo());
```
... and encountered a ParseError. Note that `node foo.js` executes it without error. Note that the error only occurs when `'use strict';` is present.

**What did you expect to happen?**

Successful linting without a Parsing Error.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$ eslint foo.js

/home/dfike/foo/foo.js
  3:23  error  Parsing error: Identifier '_foo' has already been declared

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
I can reproduce this, thanks for reporting.

This is actually a bug in [espree](https://github.com/eslint/espree). I'm working on a fix now.

**Comment by danfike** [comment]
@not-an-aardvark Thanks for digging into it so quickly!

**Comment by not-an-aardvark** [comment]
This has been fixed in https://github.com/ternjs/acorn/commit/2355566f1cfb26e9422b572a9fc25711093cba4e, so it should no longer be a problem if you purge `node_modules` and reinstall.
